kafka-client/src/test/java/dev/responsive/kafka/internal/db/partitioning/SegmenterTest.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2024 Responsive Computing, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.responsive.kafka.internal.db.partitioning;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInRelativeOrder;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+
+import dev.responsive.kafka.internal.utils.TableName;
+import java.util.Collections;
+import java.util.stream.LongStream;
+import org.junit.jupiter.api.Test;
+
+public class SegmenterTest {
+
+  private static final TableName NAME = new TableName("table");
+  private static final String CHANGELOG_TOPIC_NAME = "changelog";
+
+  @Test
+  public void shouldReturnCorrectSegmentStartTimestamp() {
+    // Given:
+    final var retentionPeriod = 10_000;
+    final var segmentIntervalMs = 1_000;
+    final var segmenter = new Segmenter(retentionPeriod, segmentIntervalMs);
+
+    // Then:
+    assertThat(segmenter.segmentStartTimestamp(0), equalTo(0L));
+    assertThat(segmenter.segmentStartTimestamp(500), equalTo(0L));
+    assertThat(segmenter.segmentStartTimestamp(999), equalTo(0L));
+    assertThat(segmenter.segmentStartTimestamp(1_001), equalTo(1_000L));
+    assertThat(segmenter.segmentStartTimestamp(2_000), equalTo(2_000L));
+  }
+
+  @Test
+  public void shouldReturnCorrectActiveSegments() {
+    // Given:
```

Comment:
Huzzah! You've adopted our given/when/then pattern and our naming schemes ❤️ I feel strongly that it encourages really nice unit test hygiene.